File: app/src/main/java/com/cyl/musiclake/ui/music/model/Music.java
package com.cyl.musiclake.ui.music.model;

import android.graphics.Bitmap;
import android.os.Parcel;
import android.os.Parcelable;

/**
 * 作者：yonglong on 2016/8/9 10:50
 * 邮箱：643872807@qq.com
 * 版本：2.5
 */
public class Music implements Parcelable {
    // 歌曲类型 本地/网络
    private Type type;
    // [本地歌曲]歌曲id
    private long id;
    // 音乐标题
    private String title;
    // 艺术家
    private String artist;
    // 专辑
    private String album;
    // 专辑id
    private long artistId;
    // 专辑id
    private long albumId;
    // 持续时间
    private long duration;
    // 音乐路径
    private String uri;
    // [网络歌曲] 音乐歌词地址
    private String lrcPath;
    // [本地歌曲]专辑封面路径
    private String coverUri;
    // 文件名
    private String fileName;
    // [网络歌曲]专辑封面bitmap
    private Bitmap cover;
    // 文件大小
    private long fileSize;
    // 发行日期
    private String year;

    public Music() {
        this.id = -1;
        this.albumId = -1;
        this.artistId = -1;

        this.title = "未知";
        this.artist = "未知";
        this.album = "未知";

        this.uri = "";
        this.lrcPath = "";
        this.coverUri = "";
        this.fileSize = -1;
        this.duration = -1;
        this.cover = null;

        this.year = "未知";
        this.type = Type.LOCAL;
    }

    protected Music(Parcel in) {
        id = in.readLong();
        title = in.readString();
        artist = in.readString();
        album = in.readString();
        artistId = in.readLong();
        albumId = in.readLong();
        duration = in.readLong();
        uri = in.readString();
        lrcPath = in.readString();
        coverUri = in.readString();
        fileName = in.readString();
        cover = in.readParcelable(Bitmap.class.getClassLoader());
        fileSize = in.readLong();
        year = in.readString();
    }

    public static final Creator<Music> CREATOR = new Creator<Music>() {
        @Override
        public Music createFromParcel(Parcel in) {
            return new Music(in);
        }

        @Override
        public Music[] newArray(int size) {
            return new Music[size];
        }
    };

    public long getArtistId() {
        return artistId;
    }

    public void setArtistId(long artistId) {
        this.artistId = artistId;
    }

    public long getAlbumId() {
        return albumId;
    }

    public void setAlbumId(long albumId) {
        this.albumId = albumId;
    }

    public String getAlbum() {
        return album;
    }

    public void setAlbum(String album) {
        this.album = album;
    }

    public String getArtist() {
        return artist;
    }

    public void setArtist(String artist) {
        this.artist = artist;
    }

    public Bitmap getCover() {
        return cover;
    }

    public void setCover(Bitmap cover) {
        this.cover = cover;
    }

    public String getCoverUri() {
        return coverUri;
    }

    public void setCoverUri(String coverUri) {
        this.coverUri = coverUri;
    }

    public long getDuration() {
        return duration;
    }

    public void setDuration(long duration) {
        this.duration = duration;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public long getFileSize() {
        return fileSize;
    }

    public void setFileSize(long fileSize) {
        this.fileSize = fileSize;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTitle() {
        return title ;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Type getType() {
        return type;
    }

    public void setType(Type type) {
        this.type = type;
    }

    public String getUri() {
        return uri;
    }

    public void setUri(String uri) {
        this.uri = uri;
    }

    public String getYear() {
        return year;
    }

    public void setYear(String year) {
        this.year = year;
    }

    public String getLrcPath() {
        return lrcPath;
    }

    public void setLrcPath(String lrcPath) {
        this.lrcPath = lrcPath;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeLong(id);
        parcel.writeString(title);
        parcel.writeString(artist);
        parcel.writeString(album);
        parcel.writeLong(artistId);
        parcel.writeLong(albumId);
        parcel.writeLong(duration);
        parcel.writeString(uri);
        parcel.writeString(lrcPath);
        parcel.writeString(coverUri);
        parcel.writeString(fileName);
        parcel.writeParcelable(cover, i);
        parcel.writeLong(fileSize);
        parcel.writeString(year);
    }

    public enum Type {
        ONLINE,
        LOCAL
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Music music = (Music) o;

        if (id != music.id) return false;
        if (artistId != music.artistId) return false;
        if (albumId != music.albumId) return false;
        if (duration != music.duration) return false;
        if (fileSize != music.fileSize) return false;
        if (type != music.type) return false;
        if (title != null ? !title.equals(music.title) : music.title != null) return false;
        if (artist != null ? !artist.equals(music.artist) : music.artist != null) return false;
        if (album != null ? !album.equals(music.album) : music.album != null) return false;
        if (uri != null ? !uri.equals(music.uri) : music.uri != null) return false;
        if (lrcPath != null ? !lrcPath.equals(music.lrcPath) : music.lrcPath != null) return false;
        if (coverUri != null ? !coverUri.equals(music.coverUri) : music.coverUri != null)
            return false;
        if (fileName != null ? !fileName.equals(music.fileName) : music.fileName != null)
            return false;
        if (cover != null ? !cover.equals(music.cover) : music.cover != null) return false;
        return year != null ? year.equals(music.year) : music.year == null;

    }

    @Override
    public int hashCode() {
        int result = type != null ? type.hashCode() : 0;
        result = 31 * result + (int) (id ^ (id >>> 32));
        result = 31 * result + (title != null ? title.hashCode() : 0);
        result = 31 * result + (artist != null ? artist.hashCode() : 0);
        result = 31 * result + (album != null ? album.hashCode() : 0);
        result = 31 * result + (int) (artistId ^ (artistId >>> 32));
        result = 31 * result + (int) (albumId ^ (albumId >>> 32));
        result = 31 * result + (int) (duration ^ (duration >>> 32));
        result = 31 * result + (uri != null ? uri.hashCode() : 0);
        result = 31 * result + (lrcPath != null ? lrcPath.hashCode() : 0);
        result = 31 * result + (coverUri != null ? coverUri.hashCode() : 0);
        result = 31 * result + (fileName != null ? fileName.hashCode() : 0);
        result = 31 * result + (cover != null ? cover.hashCode() : 0);
        result = 31 * result + (int) (fileSize ^ (fileSize >>> 32));
        result = 31 * result + (year != null ? year.hashCode() : 0);
        return result;
    }

    @Override
    public String toString() {
        return "Music{" +
                "album='" + album + '\'' +
                ", type=" + type +
                ", id=" + id +
                ", title='" + title + '\'' +
                ", artist='" + artist + '\'' +
                ", albumId=" + albumId +
                ", duration=" + duration +
                ", uri='" + uri + '\'' +
                ", coverUri='" + coverUri + '\'' +
                ", fileName='" + fileName + '\'' +
                ", cover=" + cover +
                ", fileSize=" + fileSize +
                ", year='" + year + '\'' +
                '}';
    }
}

File: app/src/main/java/com/cyl/musiclake/bean/Music.kt
package com.cyl.musiclake.bean

import android.os.Parcel
import android.os.Parcelable
import com.cyl.musicapi.playlist.QualityBean
import org.litepal.crud.LitePalSupport

/**
 * 作者：yonglong on 2016/8/9 10:50
 * 邮箱：643872807@qq.com
 * 版本：2.5
 */
class Music() : LitePalSupport(), Parcelable {
    // 歌曲类型 本地/网络
    var type: String? = null
    //数据库存储id
    var id: Long = 0
    // 歌曲id
    var mid: String? = null
    // 音乐标题
    var title: String? = null
    // 艺术家
    var artist: String? = null//{123,123,13}
    // 专辑
    var album: String? = null
    // 专辑id
    var artistId: String? = null//{123,123,13}
    // 专辑id
    var albumId: String? = null
    // 专辑内歌曲个数
    var trackNumber: Int = 0
    // 持续时间
    var duration: Long = 0
    // 收藏
    var isLove: Boolean = false
    // [本地|网络]
    var isOnline: Boolean = true
    // 音乐路径
    var uri: String? = null
    // [本地|网络] 音乐歌词地址
    var lyric: String? = null
    // [本地|网络]专辑封面路径
    var coverUri: String? = null
    // [网络]专辑封面
    var coverBig: String? = null
    // [网络]small封面
    var coverSmall: String? = null
    // 文件名
    var fileName: String? = null
    // 文件大小
    var fileSize: Long = 0
    // 发行日期
    var year: String? = null
    //更新日期
    var date: Long = 0
    //在线歌曲是否限制播放，false 可以播放
    var isCp: Boolean = false
    //在线歌曲是否付费歌曲，false 不能下载
    var isDl: Boolean = true
    //收藏id
    var collectId: String? = null
    //音乐品质，默认标准模式
    var quality: Int = 128000

    //音乐品质选择
    var hq: Boolean = false //192
    var sq: Boolean = false //320
    var high: Boolean = false //999
    //是否有mv 0代表无，1代表有
    var hasMv: Int = 0

    constructor(parcel: Parcel) : this() {
        type = parcel.readString()
        id = parcel.readLong()
        mid = parcel.readString()
        title = parcel.readString()
        artist = parcel.readString()
        album = parcel.readString()
        artistId = parcel.readString()
        albumId = parcel.readString()
        trackNumber = parcel.readInt()
        duration = parcel.readLong()
        isLove = parcel.readByte() != 0.toByte()
        isOnline = parcel.readByte() != 0.toByte()
        uri = parcel.readString()
        lyric = parcel.readString()
        coverUri = parcel.readString()
        coverBig = parcel.readString()
        coverSmall = parcel.readString()
        fileName = parcel.readString()
        fileSize = parcel.readLong()
        year = parcel.readString()
        date = parcel.readLong()
        isCp = parcel.readByte() != 0.toByte()
        isDl = parcel.readByte() != 0.toByte()
        collectId = parcel.readString()
        quality = parcel.readInt()
        hq = parcel.readByte() != 0.toByte()
        sq = parcel.readByte() != 0.toByte()
        high = parcel.readByte() != 0.toByte()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(p0: Parcel, p1: Int) {
        p0.writeString(type)
        p0.writeLong(id)
        p0.writeString(mid)
        p0.writeString(title)
        p0.writeString(artist)
        p0.writeString(album)
        p0.writeString(artistId)
        p0.writeString(albumId)
        p0.writeInt(trackNumber)
        p0.writeLong(duration)
        p0.writeByte(if (isLove) 1 else 0)
        p0.writeByte(if (isOnline) 1 else 0)
        p0.writeString(uri)
        p0.writeString(lyric)
        p0.writeString(coverUri)
        p0.writeString(coverBig)
        p0.writeString(coverSmall)
        p0.writeString(fileName)
        p0.writeLong(fileSize)
        p0.writeString(year)
        p0.writeLong(date)
        p0.writeByte(if (isCp) 1 else 0)
        p0.writeByte(if (isDl) 1 else 0)
        p0.writeString(collectId)
        p0.writeInt(quality)
        p0.writeByte(if (hq) 1 else 0)
        p0.writeByte(if (sq) 1 else 0)
        p0.writeByte(if (high) 1 else 0)
    }

    override fun toString(): String {
        return "Music(type=$type, id=$id, mid=$mid, title=$title, " +
                "artist=$artist, album=$album, artistId=$artistId, " +
                "albumId=$albumId, trackNumber=$trackNumber," +
                " duration=$duration, isLove=$isLove, isOnline=$isOnline, " +
                "uri=$uri, lyric=$lyric, coverUri=$coverUri, coverBig=$coverBig, coverSmall=$coverSmall," +
                " fileName=$fileName, fileSize=$fileSize, year=$year, date=$date, isCp=$isCp, isDl=$isDl, " +
                "collectId=$collectId, quality=$quality," +
                "qualityList=$high $hq $sq)"
    }

    companion object CREATOR : Parcelable.Creator<Music> {
        override fun createFromParcel(parcel: Parcel): Music {
            return Music(parcel)
        }

        override fun newArray(size: Int): Array<Music?> {
            return arrayOfNulls(size)
        }
    }
}

